### ISSUE TITLE
Generate spec looks up types in wrong package if there are several of the same name

### ISSUE THREAD

**Issue by bfirsh** [go-swagger/go-swagger#608]
The message introduced in #437 is causing pretty sporadic error messages for me. Every time I run`swagger generate spec`, I get different paths for the error:

```
package github.com/coreos/etcd/pkg/types, error is: unknown builtin "Port"
```

```
package github.com/docker/containerd/api/grpc/types, error is: unknown builtin "Port"
```

... and so on. None of those packages have anything to do with the type "Port", nor have any swagger definitions in them.

I'm running off master.

**Comment by casualjim** [comment]
Are the sources for what you're trying to run the generate spec against available? 
I'd like to take a look at the project and be able to reproduce the issue

**Comment by bfirsh** [comment]
https://github.com/bfirsh/docker/tree/generate-swagger-specification
https://github.com/bfirsh/engine-api/tree/add-swagger-annotations

Note that it currently doesn't generate at all for reasons I can't figure out (`Port` is defined as a model, but it's still throwing an error for some reason). I've been running a version of go-swagger which ignores unknown types to try and debug: https://github.com/bfirsh/go-swagger/commit/5acf8482e5cd05a7f0fe4bec9155c7b063f41031

**Comment by bfirsh** [comment]
This is quite hard to debug because the errors seem somewhat non-deterministic, but it seems the object might incorrect in this error too. With that patch applied to ignore unknown types, `Port` _sometimes_ shows up in the output, and _sometimes_ doesn't. If I disable some of the more complex routes (e.g. creating containers), then `Port` consistently shows up in the output.

Perhaps there is an object that is causing the error that isn't actually mentioned in the error? Perhaps files are getting processed in a different order, so `Port` sometimes gets shown in the output, but sometimes doesn't? Not sure... but might be a clue!

**Comment by casualjim** [comment]
I won't be able to dig into this today, but I will for sure over the weekend.

**Comment by bfirsh** [comment]
Fantastic, thanks!

This root problem causing the error is #241, I believe. It also hints at some of the non-deterministic error.

It's quite possible I'm just doing something wrong, but the error message isn't pointing me in the right place so it's very hard to figure out what is actually wrong.

**Comment by casualjim** [comment]
I have an error about github.com/notary/tuf/store when I get the source of that I don't have the store package.
Is there a particular branch I should be looking at?

**Comment by bfirsh** [comment]
Docker's pretty hairy to build, so perhaps this isn't the demonstration of the problem. I might try and build a smaller reproducible test case.

If you want to dive further into this, this is the command I'm running to get a build environment set up:

```
DOCKER_MOUNT="-v $(pwd):/go/src/github.com/docker/docker -v $(pwd)/../engine-api:/go/src/github.com/docker/engine-api" shell
```

That builds a container with all of the dependencies in there. To then build the swagger definition:

```
hack/make.sh swagger
```

**Comment by bfirsh** [comment]
I'm running into a similar issue now with a different object:

    unable to find NetworkCreate in github.com/coreos/etcd/pkg/types

Here's the relevant code: https://github.com/bfirsh/engine-api/blob/8b2ebf69c454c8b2122d6e16a1a0e75bf59d12f9/types/types.go#L507

`NetworkCreate` is used in `NetworkCreateRequest` and `NetworkCreateResponse` which are the actual objects used for Swagger. The obvious weird thing is that is that `github.com/coreos/etcd/pkg/types` is the wrong package. I've no idea why it's picking that package, because it doesn't have a `NetworkCreate` object or anything.

The other weird thing is why it doesn't work in the first place. I'm guessing there is a real error that is being swallowed somewhere, but I'm not sure where to look.

**Comment by bfirsh** [comment]
Ahah - I've got it! The clue is that the package names are all called `types`. I have several packages called `types`, and `packageForFile` looks for packages by package name only, so it is picking the wrong one.

I shall investigate further...

**Comment by bfirsh** [comment]
A half fix! #623

**Comment by casualjim** [comment]
I was able to generate a swagger spec in a docker shell with the code that is currently on scan-package branch. You'll see lots of output because the ./make.sh swagger bit has an export DEBUG=1
